src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -446,6 +449,7 @@ void PEAState::add_new_allocation(Node* obj) {
     } else if (PEA_debug_idx < 0 && alloc->_idx == static_cast<node_idx_t>(-PEA_debug_idx)) { // block PEA_debug_idx
       return;
     }
+    ++peaNumAllocsTracked;
```

Comment:
we need Atomic::inc(&peaNumAllocsTracked) for them because there are multiple c2 threads.

Reply:
I made this change. But its worth noting, that other stats in `Parse::print_statistics()` such as `nodes_created`, do not have locking or atomic operations.

Reply:
yes. they are there before atomic.hpp is added.